### PR TITLE
HIVE-24726: Track required data for cache hydration

### PR DESCRIPTION
**Pull request by asinkovits**
### What changes were proposed in this pull request?
This is a subtask for the cache hydration feature, collecting the required data for storing and restoring the cache content.

### Why are the changes needed?
LLAP cache hydration will enable save/load the cache contents. To do this some additional metadata is required.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

Unit tests were added.

### CODE REVIEW

File: llap-server/src/java/org/apache/hadoop/hive/llap/cache/MemoryLimitedPathCache.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  @Override
+  public String resolve(Object key) {
+    return key != null ? internalCache.getIfPresent(key) : null;
+  }
+
+  private static class PathWeigher implements Weigher<Object, String> {
+
+    @Override
+    public int weigh(Object key, String value) {
+      // String memory footprint
```
Comment (szlta) [inline llap-server/src/java/org/apache/hadoop/hive/llap/cache/MemoryLimitedPathCache.java:55]: Please add more comments on why this is calculated like this.

File: llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java [hunk 2]
```diff
@@ -112,6 +115,8 @@
   private final BufferUsageManager bufferManager;
   private final Configuration daemonConf;
   private final LowLevelCacheMemoryManager memoryManager;
+  private LowLevelCachePolicy realCachePolicy;
```
Comment (szlta) [inline llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java:118]: Why was this refactor necessary? (similarly why for tracePool)

**Review by szlta — COMMENTED** [review]
Looks good! Minor comments only.

File: llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java (hunk 2 above)
Comment (asinkovits) [reply llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java]: Both will be required in HIVE-24727, I've reverted them for now.

File: llap-server/src/java/org/apache/hadoop/hive/llap/cache/MemoryLimitedPathCache.java (hunk 1 above)
Comment (asinkovits) [reply llap-server/src/java/org/apache/hadoop/hive/llap/cache/MemoryLimitedPathCache.java]: Done.

**Review by szlta — APPROVED** [review]
+1

**Pull request merged**